components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -243,9 +244,35 @@ public User createUser(User user, Map<String, Boolean> requiredAttributes)
 
             Map<String, String> claimsInLocalDialect = SCIMCommonUtils.convertSCIMtoLocalDialect(claimsMap);
 
-            // Create the user in the user core.
-            org.wso2.carbon.user.core.common.User coreUser = carbonUM.addUserWithID(user.getUserName(),
-                    user.getPassword(), null, claimsInLocalDialect, null);
+            org.wso2.carbon.user.core.common.User coreUser = null;
+            /*Provide a preferred primary login identifier.Generate a unique user id as the immutable identifier of
+             the user instead of human readable username. The primary login identifier claim value will be the
+             human readable username.*/
+            if (isLoginIdentifiersEnabled()) {
+                String immutableUserIdentifier = getUniqueUserID();
+                if (StringUtils.isNotBlank(getPrimaryLoginIdentifierClaim())) {
```

Comment:
since `getPrimaryLoginIdentifireClaim()` method is getting called in multiple places, let's assign this value to a variable and call the variable wherever needed.

Reply:
Fixed